readthedocs_build/config/test_config.py

```diff
@@ -413,6 +396,70 @@ def it_fails_if_base_does_not_exist(tmpdir):
         assert excinfo.value.code == INVALID_PATH
 
 
+def describe_validate_build():
+
+    def it_fails_if_build_is_invalid_option(tmpdir):
+        apply_fs(tmpdir, minimal_config)
+        build = BuildConfig(
+            {},
+            {'build': {'image': 3.0}},
+            source_file=str(tmpdir.join('readthedocs.yml')),
+            source_position=0)
+        with raises(InvalidConfig) as excinfo:
+            build.validate_build()
+        assert excinfo.value.key == 'build'
+        assert excinfo.value.code == INVALID_CHOICE
+
+    def it_fails_on_python_validation(tmpdir):
+        apply_fs(tmpdir, minimal_config)
+        build = BuildConfig(
+            {},
+            {
+                'build': {'image': 1.0},
+                'python': {'version': '3.3'},
+            },
+            source_file=str(tmpdir.join('readthedocs.yml')),
+            source_position=0)
+        build.validate_build()
+        with raises(InvalidConfig) as excinfo:
+            build.validate_python()
+        assert excinfo.value.key == 'python.version'
+        assert excinfo.value.code == INVALID_CHOICE
+
+    def it_works_on_python_validation(tmpdir):
+        apply_fs(tmpdir, minimal_config)
+        build = BuildConfig(
+            {},
+            {
+                'build': {'image': 'latest'},
+                'python': {'version': '3.3'},
+            },
+            source_file=str(tmpdir.join('readthedocs.yml')),
+            source_position=0)
+        build.validate_build()
+        build.validate_python()
+
+    def it_works(tmpdir):
+        apply_fs(tmpdir, minimal_config)
+        build = BuildConfig(
+            {},
+            {'build': {'image': 'latest'}},
+            source_file=str(tmpdir.join('readthedocs.yml')),
+            source_position=0)
+        build.validate_build()
+        assert build['build']['image'] == 'latest'
+
+    def default(tmpdir):
+        apply_fs(tmpdir, minimal_config)
+        build = BuildConfig(
+            {},
+            {},
+            source_file=str(tmpdir.join('readthedocs.yml')),
+            source_position=0)
+        build.validate_build()
+        assert build['build']['image'] == '2.0'
```

Comment:
Same here but 2.7, I guess.

Reply:
Not sure we want to specify python magically, based on container image, do we?

Reply:
We already do this to a certain degree, but i think the python version tests are testing some of this. For example:

* `python['version'] = 3` is valid, and exists in the dashboard, we should select the right python 3 version (pretty sure tests exist here)
* `python['version'] = None` should return `2.7` for now, until we're opinionated about forcing `3.6` default (not work i think we need to make for ourselves just yet, but i'd like to see this soon)

Reply:
If we don't want that, what would it happen if the user doesn't specify the python version? should the test fail? do we have a test for that?